lib/temporal/connection.rb

```diff
@@ -10,12 +10,17 @@ def self.generate(configuration)
       connection_class = CLIENT_TYPES_MAP[configuration.type]
       host = configuration.host
       port = configuration.port
+      credentials = configuration.credentials || :this_channel_is_insecure
 
       hostname = `hostname`
       thread_id = Thread.current.object_id
       identity = "#{thread_id}@#{hostname}"
 
-      connection_class.new(host, port, identity)
+      if configuration.type == :grpc
+        connection_class.new(host, port, identity, credentials)
```

Comment:
We can either loose the condition (since we only have one type of connection) or pass `credentials` as `options` Hash (allowing for other extensions in the future). Either way different signatures violate Liskov substitution principle (these are not technically subclasses, but you still want to maintain a consistent interface)